dial.go

```diff
@@ -19,6 +19,11 @@ import (
 	msmux "github.com/multiformats/go-multistream"
 )
 
+// DialTimeout is the maximum duration a Dial is allowed to take.
+// This includes the time between dialing the raw network connection,
+// protocol selection as well the handshake, if applicable.
+var DialTimeout = 30 * time.Second
```

Comment:
This limits the maximum ping to quite a low value. To finish handshake we need about 5 round-trips which makes max RTT be 6s.

Reply:
6s RTT is *really* long. However, I could make these finer grained (I'll go do that, actually).

Reply:
Ok, I just went ahead and bumped this to 60s (what we had before for the secio handshake). That should be sufficient, IMO.

I did not add fine-grained timeouts. I'd like to respect the context timeout as much as possible. We can mess with that later but I'd like to merge this ASAP (I've already let this whole mess go *way* too long).